### PR TITLE
Adds new blame-interaction degree plot

### PR DESCRIPTION
**Pull request by vulder**
With this new plot we can visualize how the degree of blame interaction changes over time.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/cache_helper.py  5
- varats/data/reports/blame_report.py  1
- varats/data/revisions.py  1
- varats/plots/blame_interaction_degree.py  6
- varats/plots/plot.py  1
         

Clones added
============
- varats/plots/commit_interactions.py  1
- varats/plots/blame_interaction_degree.py  2
         

Clones removed
==============
+ varats/plots/paper_config_overview.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4465769&bid=15082522)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  1
- varats/plots/blame_interaction_degree.py  7
- varats/data/cache_helper.py  5
- varats/plots/plot.py  1
- varats/data/revisions.py  1
         

Clones added
============
- varats/plots/blame_interaction_degree.py  3
- varats/plots/commit_interactions.py  2
         

Clones removed
==============
+ varats/plots/paper_config_overview.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4465769&bid=15082522)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/plots/plot.py  1
- varats/data/cache_helper.py  5
- varats/plots/blame_interaction_degree.py  7
- varats/data/reports/blame_report.py  1
- varats/data/revisions.py  1
         

Clones added
============
- varats/plots/blame_interaction_degree.py  2
- varats/plots/commit_interactions.py  1
         

Clones removed
==============
+ varats/plots/paper_config_overview.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4465769&bid=15082522)

File: varats/data/reports/blame_report.py [hunk 1]
```diff
@@ -165,3 +173,20 @@ def __str__(self) -> str:
         for function in self.__function_entries.values():
             str_representation += str(function) + "\n"
         return str_representation
+
+
+def generate_degree_tuples(report: BlameReport) -> tp.List[tp.Tuple[int, int]]:
+    """
+    Generates a list of tuples(degree, amount) where degree is the interaction
+    degree of a blame interaction, e.g., the number of incomming interactions,
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
    degree of a blame interaction, e.g., the number of incoming interactions,
```

File: varats/data/reports/blame_report.py [hunk 2]
```diff
@@ -165,3 +173,20 @@ def __str__(self) -> str:
         for function in self.__function_entries.values():
             str_representation += str(function) + "\n"
         return str_representation
+
+
+def generate_degree_tuples(report: BlameReport) -> tp.List[tp.Tuple[int, int]]:
+    """
+    Generates a list of tuples(degree, amount) where degree is the interaction
```
Comment (boehmseb) [inline varats/data/reports/blame_report.py]:
```suggestion
    Generates a list of tuples (degree, amount) where degree is the interaction
```

File: varats/data/revisions.py [hunk 3]
```diff
@@ -100,6 +100,39 @@ def get_processed_revisions(project_name: str,
...
+        file_name_filter: tp.Optional[tp.Callable[[str], bool]] = None
+) -> tp.List[Path]:
+    """
+    Returns a list of file paths to correctly processed revision files.
+
+    Args:
+        project_name: target project
+        result_dir: folder to look for result files
+        result_file_type: the type of the result file
+        file_name_filter: optional filter to exclude certaint files,
```
Comment (boehmseb) [inline varats/data/revisions.py]:
```suggestion
        file_name_filter: optional filter to exclude certain files,
```

File: varats/plots/blame_interaction_degree.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    def create_dataframe_layout() -> pd.DataFrame:
+        df_layout = pd.DataFrame(
+            columns=['revision', 'degree', 'amount', 'fraction'])
+        df_layout.degree = df_layout.degree.astype('int64')
+        df_layout.amount = df_layout.amount.astype('int64')
+        df_layout.fraction = df_layout.fraction.astype('int64')
+        return df_layout
+
+    def create_data_frame_for_report(report: BlameReport) -> pd.DataFrame:
+        list_of_degree_occurences = generate_degree_tuples(report)
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
        list_of_degree_occurrences = generate_degree_tuples(report)
```

File: varats/plots/blame_interaction_degree.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+        degrees, amounts = map(list, zip(*list_of_degree_occurences))
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
        degrees, amounts = map(list, zip(*list_of_degree_occurrences))
```

File: varats/plots/blame_interaction_degree.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            'linewidth': 1 if view_mode else 0.25,
+            'legend_size': 8 if view_mode else 2,
+            'xtick_size': 10 if view_mode else 2
+        }
+
+        style.use(self.style)
+
+        def cs_filter(data_frame: pd.DataFrame) -> pd.DataFrame:
+            """
+            Filter out all commit that are not in the case study, if one was
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
            Filter out all commits that are not in the case study, if one was
```

File: varats/plots/blame_interaction_degree.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            selected. This allows us to only load file related to the
```
Comment (boehmseb) [inline varats/plots/blame_interaction_degree.py]:
```suggestion
            selected. This allows us to only load files related to the
```

**Review by boehmseb — APPROVED** [review]

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- varats/data/reports/blame_report.py  1
- varats/plots/blame_interaction_degree.py  7
- varats/data/cache_helper.py  5
- varats/plots/plot.py  1
         

Clones added
============
- varats/plots/blame_interaction_degree.py  1
         

Clones removed
==============
+ varats/plots/paper_config_overview.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=4465769&bid=15082522)

**Pull request merged**